cluster-provision/images/vm-image-builder/publish-multiarch-containerdisk.sh

```diff
@@ -87,8 +104,9 @@ build_image() {
 publish_image() {
     local full_image_name="${1:?}"
     for arch in ${archs[*]};do
-        docker tag ${build_target}:devel-${arch} ${full_image_name}-${arch}
-        skopeo copy "docker-daemon:${full_image_name}-${arch}" "docker://${full_image_name}-${arch}"
+        local arch_tag=$(format_arch_name ${arch})
+        docker tag ${build_target}:devel-${arch} ${full_image_name}-${arch_tag}
```

Comment:
The use of (for example) `-arm64` instead of `-aarch64` when pushing makes the documentation introduced earlier inaccurate.

I really dislike the fact that, with this setup, we build images as `-aarch64` but push them as `-arm64`. And that, in general, we mix Go-style and Linux-style architecture names in user-visible places.

We should be able to use Go-style architecture names pretty much everywhere: we could use `image-url-arm64` instead of `image-url-aarch64`, pass `ARCHITECTURE=arm64` to the various scripts, and so on. Even buildx's `--platform` option takes Go-style architecture names.

AFAICT the only place where we really need to use Linux-style architecture names is the virt-install command line. We should convert from the provided Go-style architecture name to the Linux-style architecture name just there.

Reply:
updated